### PR TITLE
[TR] PIM-4973: Fix product removal from PEF (in mongo)

### PR DESCRIPTION
**Pull request by phaseinducer**
https://akeneo.atlassian.net/browse/PIM-4973

| Q                 | A
| ----------------- | ---
| Specs             | n/a
| Behats            | yes
| Blue CI           | ok
| Changelog updated | yes
| Review and 2 GTM  | ok

### CODE REVIEW

**Comment by skeleton** [comment]
GTM

File: features/product/remove_product.feature [hunk 1]
```diff
@@ -20,15 +20,17 @@ Feature: Remove a product
...
   Scenario: Successfully delete a product from the grid
     Given I am on the products page
-    Then I should see products boots
+    Then I should see product boots
     When I click on the "Delete the product" action of the row which contains "boots"
     Then I should see "Delete confirmation"
-    And I confirm the removal
-    Then I should not see product CD player
+    When I confirm the removal
+    Then I should see "Product successfully removed"
```
Comment (nidup) [inline features/product/remove_product.feature]: tests on flash message are really unstable, we avoid to use them

File: features/product/remove_product.feature [hunk 2]
```diff
@@ -20,15 +20,17 @@ Feature: Remove a product
...
+    And I should not see product boots
 
   Scenario: Successfully delete a product from the edit form
     Given I am on the "boots" product page
     And I press the "Delete" button
     Then I should see "Confirm deletion"
-    And I confirm the removal
-    Then I should not see product boots
+    When I confirm the removal
+    Then I should see "Product successfully deleted"
```
Comment (nidup) [inline features/product/remove_product.feature]: tests on flash message are really unstable, we avoid to use them

**Comment by nidup** [comment]
baby comment about flash message then GTM

**Comment by ClementGautier** [comment]
GTM

**Pull request merged**